selftests/unit/test_utils_asset.py

```diff
@@ -102,6 +102,32 @@ def test_unknown_scheme(self):
                               None, None, None, [self.cache_dir], None)
         self.assertRaises(asset.UnsupportProtocolError, invalid.fetch)
 
+    def test_fetch_two_files_same_name_no_hash(self):
+        """
```

Comment:
this makes sense, but it's quite late. I'll think about it over night and get back to you tomorrow.